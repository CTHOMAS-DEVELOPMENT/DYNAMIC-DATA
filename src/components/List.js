import React from "react";
import { connect } from "react-redux";
export const ConnectedList = props => {
  let gifts = [];
  //Get the gifts derived from the 'Main' component
  if (props.gifts !== undefined) {
    gifts = props.gifts;
  }
  const formatTextToNumber=(textNumber)=>
{
  return textNumber.toString().replace(/(\d)(?=(\d\d\d)+(?!\d))/g, "$1,")
}
  const block={ display:"inlineBlock" }
  return (
    <div >
      {gifts.length === 0 ? (
        <h2>No Transactions</h2>
      ) : (
        <div >
          <h2>Transactions</h2>
          <div className="list-group list-group-flush">
            {gifts.map((gift,id) => {
              //console.log("gift",gift)
              return (
                <div className="list-group-item" key={id}>
                  <div style={block}>{gift[1].name}</div>
                  <div style={block}>
                    {gift[1].email} <b>&pound;{formatTextToNumber(gift[1].amount)}</b> 
                  </div>
                </div>
              );
            })}
          </div>
        </div>
      )}
     
    </div>
  );
};
const mapStateToProps = state => {
    return { gifts: state.gifts };
  };

export default connect(mapStateToProps)(ConnectedList);
