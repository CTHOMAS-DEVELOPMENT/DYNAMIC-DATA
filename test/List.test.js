import React from "react";
import { shallow } from "enzyme";
import List from "../src/components/List";
import { Provider } from "react-redux";
import store from "../src/store"

test("Should render List component correctly..", () => {
  const wrapper = shallow(
    <Provider store={store}>
    <List gifts={[]} />
    </Provider>);
  expect(wrapper).toMatchSnapshot();
});
