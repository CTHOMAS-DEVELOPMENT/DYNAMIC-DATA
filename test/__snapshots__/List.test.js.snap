// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render List component correctly.. 1`] = `
<Connect(ConnectedList)
  gifts={Array []}
/>
`;
